pkg/operation/botanist/component/istio/ingress_gateway.go

```diff
@@ -111,3 +121,280 @@ func addSuffixToManifestsName(charts *chartrenderer.RenderedChart, suffix string
 		charts.Manifests[i].Name = charts.Manifests[i].Name + "/" + suffix + ".yaml"
 	}
 }
+
+// NetworkPolicyValues contains deployment parameters for the istio-ingress network policies.
+type NetworkPolicyValues struct {
+	// NodeLocalIPVSAddress is the CIDR of the node-local IPVS address.
+	NodeLocalIPVSAddress *string
+	// DNSServerAddress is the CIDR of the usual DNS server address.
+	DNSServerAddress *string
+}
+
+type networkPolicyTransformer struct {
+	name      string
+	transform func(*networkingv1.NetworkPolicy) func() error
+}
+
+func getIstioNetworkPolicyTransformers(values NetworkPolicyValues) []networkPolicyTransformer {
+	return []networkPolicyTransformer{
+		{
+			name: "deny-all-egress",
+			transform: func(obj *networkingv1.NetworkPolicy) func() error {
+				return func() error {
+					obj.Annotations = map[string]string{
+						v1beta1constants.GardenerDescription: "Deny all egress traffic from pods in this namespace.",
+					}
+					obj.Spec = networkingv1.NetworkPolicySpec{
+						PolicyTypes: []networkingv1.PolicyType{networkingv1.PolicyTypeEgress},
+					}
+					return nil
+				}
+			},
+		},
+		{
+			name: "to-shoot-apiserver",
+			transform: func(obj *networkingv1.NetworkPolicy) func() error {
+				return func() error {
+					obj.Annotations = map[string]string{
+						v1beta1constants.GardenerDescription: fmt.Sprintf("Allows Egress from pods labeled with "+
+							"'%s=%s' to shoot api servers with label '%s=%s'.", v1beta1constants.LabelApp, v1beta1constants.DefaultIngressGatewayAppLabelValue,
+							v1beta1constants.GardenRole, v1beta1constants.GardenRoleControlPlane),
```

Comment:
This is not correct. The network policy does not allows communication to all `controlplane` pods. Please replace it with something more specific, e.g.
```suggestion
							v1beta1constants.LabelRole, v1beta1constants.LabelAPIServer),
```